mitodl/mitxonline
# 905: dashboard overflow menu ⋮ accessibility

#### Pre-Flight checklist

#### What are the relevant tickets?
https://github.com/mitodl/mitxonline/issues/905
https://github.com/mitodl/mitxonline/issues/909

#### What's this PR do?
Reorganizes the order of items in the course and program components shown on a user's Dashboard (http://<BASE_URL>/dashboard/).  The reorganization of items in each component allows for an alternate order of focus as the user presses `tab` in each component.  

Show below is the new tab order for each component on the dashboard (program component and course component)
![Screen Shot 2022-08-30 at 7 56 34](https://user-images.githubusercontent.com/8311573/187431320-3ad01c22-00a4-4269-8e94-385bfb6d8a3a.png)

This PR also changes the name and tooltip shown when hovering over the vertical-dot menu for the course component on the Dashboard.  This should also influence screen readers.

This PR also adds back the EnrollmentRoleTag which was accidentally removed from the course component in a previous PR.

This PR should also remove the vertical-dot menu from the program component shown on the dashboard (refer to https://github.com/mitodl/mitxonline/issues/909)

#### How should this be manually tested?

1. Enroll in a Program and Course.
2. Visit the Dashboard (http://<BASE_URL>/dashboard/)
3. Before clicking anywhere on the Dashboard with your cursor, begin pressing `tab` on the keyboard and ensure the order shown in the screenshot above is followed.
4. Hover your cursor over each vertical dot menu and ensure the tooltip shows `Course options for {course title}`.
5. Using a screen reader, ensure that the tooltip text described in step 4 is also audible.
